metrics/line-numbers/fineLineNumber.go

```diff
@@ -0,0 +1,48 @@
+package linenumbers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strconv"
+
+	"gopkg.in/yaml.v3"
+)
+
+func recursiveFindLine(node *yaml.Node, keyIndex int, maxDepth int, keys []string) (*yaml.Node, error) {
```

Comment:
I don't think it's necessary to include "recursive" in the names of recursive functions. Since this function is returning a node and not a line number, I think "findNode" would be a good name to use for this.